spec/models/course/condition/assessment_ability_spec.rb

```diff
@@ -13,5 +13,42 @@
 
       it { is_expected.to be_able_to(:manage, condition) }
     end
+
+    context 'when the user is a Course Student' do
+      let(:user) { create(:course_student, :approved, course: course).user }
+
+      context 'when the assessment has not started' do
+        let(:unopened_assessment) { create(:assessment, :unopened, course: course) }
+
+        it { is_expected.to_not be_able_to(:attempt, unopened_assessment) }
+      end
+
+      context 'when the assessment has started' do
+        let(:assessment1) { create(:assessment, course: course) }
+        let(:assessment2) { create(:assessment, course: course) }
+        let!(:condition) do
+          create(:assessment_condition,
+                 course: course,
+                 assessment: assessment2,
+                 conditional: assessment1)
+        end
+
+        context 'when the assessment does not have any condition' do
+          it { is_expected.to be_able_to(:attempt, assessment2) }
+        end
+
+        context "when the user does not satisfied the assessment's conditions" do
+          it { is_expected.to_not be_able_to(:attempt, assessment1) }
+        end
+
+        context "when the user satisfied the assessment's conditions" do
+          it 'is able to attempt the assessment' do
+            allow(assessment1).to receive(:conditions_satisfied_by?).and_return(true)
+
+            expect(ability.can?(:attempt, assessment1)).to be_truthy
```

Comment:
Instead of having `subject(:ability) { Ability.new(user) }` at the top, you can remove that change, and just do the following here: 
```ruby
expect(subject.can?(:attempt, assessment1)).to be_truthy
```